src/control/lib/control/pool_test.go

```diff
@@ -1117,3 +1118,297 @@ func TestControl_ListPools(t *testing.T) {
+							MockStorageConfig: MockStorageConfig{
+								TotalBytes: uint64(3) * uint64(humanize.TByte),
+								AvailBytes: uint64(2) * uint64(humanize.TByte),
+							},
+							Rank: 3,
+						},
+					},
+				},
+			},
+			ExpectedOutput: ExpectedOutput{
+				ScmBytes:  uint64(50) * uint64(humanize.GByte),
+				NvmeBytes: uint64(700) * uint64(humanize.GByte),
+			},
+		},
+		"No NVME": {
+			HostsConfigArray: []MockHostStorageConfig{
+				{
+					HostName: "foo",
+					ScmConfig: []MockScmConfig{
+						{
+							MockStorageConfig: MockStorageConfig{
+								TotalBytes: uint64(100) * uint64(humanize.GByte),
+								AvailBytes: uint64(100) * uint64(humanize.GByte),
+							},
+						},
+					},
+					NvmeConfig: []MockNvmeConfig{},
+				},
+			},
+			ExpectedOutput: ExpectedOutput{
+				ScmBytes:  uint64(100) * uint64(humanize.GByte),
+				NvmeBytes: uint64(0),
+			},
+		},
+		"SCM:NVME ratio": {
+			HostsConfigArray: []MockHostStorageConfig{
+				{
+					HostName: "foo",
+					ScmConfig: []MockScmConfig{
+						{
+							MockStorageConfig: MockStorageConfig{
+								TotalBytes: uint64(100) * uint64(humanize.GByte),
+								AvailBytes: uint64(100) * uint64(humanize.GByte),
+							},
+						},
+					},
+					NvmeConfig: []MockNvmeConfig{
+						{
+							MockStorageConfig: MockStorageConfig{
+								TotalBytes: uint64(100) * uint64(humanize.TByte),
+								AvailBytes: uint64(100) * uint64(humanize.TByte),
+							},
+							Rank: 1,
+						},
+					},
+				},
+			},
+			ExpectedOutput: ExpectedOutput{
+				ScmBytes:  uint64(100) * uint64(humanize.GByte),
+				NvmeBytes: uint64(100) * uint64(humanize.TByte),
+			},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			log, buf := logging.NewTestLogger(t.Name())
+			defer common.ShowBufferOnFailure(t, buf)
+
+			mockInvokerConfig := &MockInvokerConfig{
+				UnaryResponse: &UnaryResponse{
+					Responses: []*HostResponse{},
+				},
+			}
+			for _, hostStorageConfig := range tc.HostsConfigArray {
+				storageScanResp := MockStorageScanResp(t,
+					hostStorageConfig.ScmConfig,
+					hostStorageConfig.NvmeConfig)
+				hostResponse := &HostResponse{
+					Addr:    hostStorageConfig.HostName,
+					Message: storageScanResp,
+				}
+				mockInvokerConfig.UnaryResponse.Responses = append(mockInvokerConfig.UnaryResponse.Responses,
+					hostResponse)
+			}
+			mockInvoker := NewMockInvoker(log, mockInvokerConfig)
+
+			scmBytes, nvmeBytes, err := GetMaxPoolSize(context.TODO(), mockInvoker)
+
+			common.AssertTrue(t, err == nil, "Expected no error")
+			common.AssertEqual(t,
+				tc.ExpectedOutput.ScmBytes,
+				scmBytes,
+				fmt.Sprintf("Invalid SCM pool size: expected=%d got=%d",
+					tc.ExpectedOutput.ScmBytes,
+					scmBytes))
+
+			common.AssertEqual(t,
+				tc.ExpectedOutput.NvmeBytes,
+				nvmeBytes,
+				fmt.Sprintf("Invalid NVME pool size: expected=%d got=%d",
+					tc.ExpectedOutput.NvmeBytes,
+					nvmeBytes))
+		})
+	}
+}
+
+func TestControl_GetMaxPoolSize_Errors(t *testing.T) {
```

Comment:
we normally test for errors and expected operation in the same table-based test

Reply:
With latest, is probably easier to factorize the code.

Following tasks will be done:
- [ ] Factorize test code